keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/SpinnakerCallAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package com.netflix.spinnaker.keel.retrofit
+
+import com.netflix.spinnaker.kork.common.Header.ACCOUNTS
+import com.netflix.spinnaker.kork.common.Header.USER
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+import java.util.concurrent.atomic.AtomicReference
+import okhttp3.Request
+import org.slf4j.MDC
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.CallAdapter.Factory
+import retrofit2.Callback
+import retrofit2.Retrofit
+
+object SpinnakerCallContexts : AtomicReference<MutableMap<Request, Map<String, String>>>(mutableMapOf())
+
+class SpinnakerCallWrapper<T>(
+  private val call: Call<T>
+) : Call<T> by call {
+
+  override fun enqueue(callback: Callback<T>) {
+    SpinnakerCallContexts.getAndUpdate {
```

Comment:
The call wrapper delegates to the original call object, except here where we take the opportunity to save the headers to the shared request contexts map.